kernel/aster-nix/src/time/clocks/system_wide.rs

```diff
@@ -0,0 +1,300 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::sync::Arc;
+use core::time::Duration;
+
+use aster_frame::{
+    arch::timer::{self, Jiffies},
+    cpu_local,
+    sync::SpinLock,
+    CpuLocal,
+};
+use aster_time::read_monotonic_time;
+use paste::paste;
+use spin::Once;
+
+use crate::time::{system_time::START_TIME_AS_DURATION, timer::TimerManager, Clock, SystemTime};
+
+struct PrivateFieldMarker;
+
+/// The Clock that reads the jiffies, and turn the counter into `Duration`.
+pub struct JiffiesClock {
+    _marker: PrivateFieldMarker,
+}
+
+/// `RealTimeClock` represents a clock that provides the current real time.
+pub struct RealTimeClock {
+    _marker: PrivateFieldMarker,
+}
+
+impl RealTimeClock {
+    /// Get the singleton of this clock.
+    pub fn get() -> &'static Arc<dyn Clock> {
```

Comment:
Return `Arc<RealTimeClock>`, which avoids dynamic dispatching of methods. If the user want to erase the type, she can always convert `Arc<RealTimeClock>` to `Arc<dyn Clock>` herself.